kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/usb/UsbModemDevice.java

```diff
@@ -197,4 +250,87 @@ private int getDigitPosition(String devName) {
             return pos;
         }
     }
+
+    private class TtyDev {
+
+        @Override
+        public String toString() {
+            return "TtyDev [portName=" + this.portName + ", interfaceNumber=" + this.interfaceNumber + "]";
+        }
+
+        @Override
+        public int hashCode() {
+            final int prime = 31;
+            int result = 1;
+            result = prime * result + getOuterType().hashCode();
+            result = prime * result + (this.portName == null ? 0 : this.portName.hashCode());
+            return result;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
```

Comment:
This is a little bit asymmetrical with the comparator. The test for equality only considers the port name. This is a trick to support removing an `TtyDev` from the internal list by its name only. The comparator, instead, considers first interface numbers.

Reply:
@cdealti Ok, I can add a comment on the comparator/equals methods.